### PR TITLE
Ingestor decorator

### PR DESCRIPTION
**Pull request by SPTKL**
simplifying ingestors using decorators, also standardize processes across different destinations

### CODE REVIEW

File: library/ingest.py [hunk 1]
```diff
@@ -38,153 +19,150 @@ def compress(self, path: str, *files, inplace: bool = True):
...
+    @translator
+    def csv(
         self,
         path: str,
         compress: bool = False,
         inplace: bool = False,
-    ) -> bool:
+    ):
         """
         https://gdal.org/drivers/vector/pgdump.html
```
Comment (mgraber) [inline library/ingest.py:119]: Remove these URLs from all of the non-PG dump functions.

**Review by mgraber — COMMENTED** [review]
Some general comments:

Do we need to add python-sqlalchemy to these ReadMe instructions?
```
sudo apt install -y gdal-bin libgdal-dev python3-gdal
```

What destination format allows for the option `OVERWRITE=YES`? Only going straight to postgres? The ingestors for csv, pgdump, geojson, and shapefile cannot use this option. [This line](https://github.com/NYCPlanning/db-data-library/blob/d9fa3e5b69ee899a58eac57e53f9e18bcc4c323a/library/ingest.py#L75) seems to suggest everything gets overwritten.

File: library/ingest.py (hunk 1 above)
Comment (SPTKL) [reply library/ingest.py:119]: I thought sqlalchemy is included in the pyproject.toml, I've added some new dependencies, you might need to run `poetry install` again

Comment (mgraber) [reply library/ingest.py:119]: Yup, I found it. Thanks!

Comment (SPTKL) [reply library/ingest.py:119]: the layer creation option is different for all different kinds of file format, we don't have a custom way of handling it yet, ideally under destination options, we should have somethjing like
```yaml
options
   - csv
      - "this=that"
   - geojson
      - "that=this"
   - postgres
      - "OVERWRITE=YES"
```

Comment (SPTKL) [reply library/ingest.py:119]: but this is lower priority, since if the file format is wrong, it would just throw a warning. we can handle them in future

Comment (SPTKL) [reply library/ingest.py:119]: the reason I included these urls         https://gdal.org/drivers/vector/pgdump.html
is to help users to find the corresponding layercreationioptions, would need to update them at one point

Comment (mgraber) [reply library/ingest.py:119]: But why do we have the link for pgdump for all functions?

Comment (SPTKL) [reply library/ingest.py:119]: oh I forgot to change them, the functions were copy and pasted

**Pull request merged**